Fix plugin helper loading conflict error

markstory opened the pull request:
When plugin helpers are lazy-loaded with a non-prefixed name we shouldn't raise errors when those helpers are also loaded via fallback helper loading. Thanks to @kolorafa for providing a simple application that helps reproduce this issue.

Fixes #17268

markstory committed:
Fix plugin helper loading conflict error

When plugin helpers are lazy-loaded with a non-prefixed name we
shouldn't raise errors when those helpers are also loaded via fallback
helper loading. Thanks to @kolorafa for providing a simple application
that helps reproduce this issue.

Fixes #17268

othercorey commented:
Should this be backported to 4.x?

markstory commented:
> Should this be backported to 4.x?

I wasn't able to reproduce the problem in 4.x

dereuromark commented:
I have the same problem now with components, though

AuthUserComponent:

	protected array $components = ['TinyAuth.Auth'];

using the Auth component internally

and AppController::initialize():

		$this->loadComponent('TinyAuth.Auth');
		$this->loadComponent('TinyAuth.AuthUser');

loading both components for use

now with exception:

> The `Auth` alias has already been loaded. The `className` was not defined in the previous configuration data.
Cake\Core\Exception\CakeException

dereuromark commented:
Hm, the problem still seems to exist even with latest 5.x code:
https://github.com/dereuromark/cakephp-data/actions/runs/6299496110/job/17100356121#step:9:73

markstory commented:
This stack trace is different than the issue I fixed. The issue I fixed was helpers being loaded by other helpers, after the view was initialized. The stacktrace you have looks like helpers being loaded during `View::__construct()`. Do you have helpers that eagerload other helpers during `__construct`?

dereuromark commented:
This is inside a test case, so there could be a än outdated way of adding them
